posts/virtual_environments.md

```diff
@@ -0,0 +1,126 @@
+
+![Figure 6: Virtual Environments end-to-end](virtual_environments/virtual_envs_end_to_end.png)
+*Figure 6: Virtual Environments end-to-end.*
+
+Views that belong to environments other than the production one get an environment name attached as a suffix to the schema portion of their fully qualified names. Thus the `db.model_a` dataset can be accessed using its original name in the production environment, while in order to access it in the environment named `test` the `db__test.model_a` name should be used instead. That's how changes can be **previewed** before making their way into production.
+
+When a new environment is created it shares the same set of snapshot table pointers with the production environment. Once changes are applied to that environment, new model snapshots are generated and some of the views get updated. Views that are part of the production environment remain unaffected. This is how SQLMesh ensures environment **isolation**.
+
+Finally, if a model implementation hasn't changed in relation to production (either **directly** or **indirectly**), the platform can safely **reuse** the same snapshot table that is currently being used in the production environment. Similarly, new snapshot tables created as part of the development process can be safely reused when changes are promoted into production.
+
+## Conclusions
+With **virtual environments** SQLMesh is able to provide fully **isolated** development environments with a complete **preview** of changes before they make it into production while **reusing** production datasets when it's appropriate and safe to do so. This, in turn, results in very tangible benefits for end users:
+- Creating new environments is cheap since it only involves creation of a new set of views.
+- Data is immediately available in new environments thanks to the separation between **virtual** and **physical** layers.
+- Changes and their downstream impact are categorized automatically to achieve both efficiency and correctness.
+- Rolling back a change happens almost instantaneously since no data movement is involved and only views that are part of the **virtual layer** get updated.
+- Deploying changes to production is also a **virtual layer** operation, which helps ensure that results observed during development are exactly the same on production and that data and code are always in sync.
+
+When it comes to the last point of deploying changes to production, our team is about to release the [CI/CD bot](https://github.com/TobikoData/sqlmesh/blob/main/docs/integrations/github.md) which will help streamline and automate this process. Don't miss out, join our [Slack channel](https://join.slack.com/t/tobiko-data/shared_invite/zt-1ma66d79v-a4dbf4DUpLAQJ8ptQrJygg) and stay tuned.
```

Comment:
"To help streamline deploying changes to production, our team[...]which will help automate this process."

Make last sentence its own paragraph so people don't miss the call to action.